Modules/ApplicationInsights/2.0/deploy.json

```diff
@@ -0,0 +1,199 @@
+      "apiVersion": "2018-02-01",
+      "name": "[variables('pidName')]",
+      "properties": {
+        "mode": "Incremental",
+        "template": {
+          "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+          "contentVersion": "1.0.0.0",
+          "resources": []
+        }
+      }
+    },
+    {
+      "type": "Microsoft.Insights/components",
+      "apiVersion": "[providers('Microsoft.Insights','components').apiVersions[0]]",
+      "name": "[parameters('appInsightsName')]",
+      "location": "[parameters('location')]",
+      "tags": {
+        "Environment": "[parameters('tagEnvironment')]",
+        "Application": "[parameters('tagApplication')]",
+        "Project": "[parameters('tagProject')]",
+        "Owner": "[parameters('tagOwner')]",
+        "OwnerEmail": "[parameters('tagOwnerEmail')]"
+      },    
+      "properties": {
+        "ApplicationId": "[parameters('appInsightsName')]",
+        "Application_Type": "[parameters('appInsightsType')]"
+      }
+    },
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "apiVersion": "[providers('Microsoft.Storage','storageAccounts').apiVersions[0]]",
+      "name": "[parameters('storageAccountName')]",
+      "location": "[resourceGroup().location]",
+      "tags": {
+        "Environment": "[parameters('tagEnvironment')]",
+        "Application": "[parameters('tagApplication')]",
+        "Project": "[parameters('tagProject')]",
+        "Owner": "[parameters('tagOwner')]",
+        "OwnerEmail": "[parameters('tagOwnerEmail')]"
+      },
+      "sku": {
+        "name": "[parameters('storageAccountSku')]"
+      },    
+      "properties": {
+        "supportsHttpsTrafficOnly": true
+      }
+    }
+  ],
+  "outputs": {
+    "appInsightsName": {
+      "type": "string",
+      "value": "[variables('appInsightsName')]",
+      "metadata": {
+        "description": "Application Insights Resource Name"
+      }
+    },
+    "appInsightsResourceId": {
+      "type": "string",
+      "value": "[variables('appInsightsresourceId')]",
+      "metadata": {
+        "description": "Application Insights Resource Id"
+      }
+    },
+    "appInsightsResourceGroup": {
+      "type": "string",
+      "value": "[resourceGroup().name]",
+      "metadata": {
+        "description": "Application Insights ResourceGroup"
+      }
+    },
+    "appInsightsKey": {
+      "type": "string",
+      "value": "[reference(variables('appInsightsresourceId'),variables('apiVersion')).instrumentationKey]",
+      "metadata": {
+        "description": "Application Insights Resource Instrumentation Key"
+      }
+    },
+    "appInsightsAppId": {
+      "type": "string",
+      "value": "[reference(variables('appInsightsresourceId'),variables('apiVersion')).AppId]",
+      "metadata": {
+        "description": "Application Insights Paalication Id"
+      }
+    },
+    "appInsightsStorageAccountName": {
```

Comment:
I think this is not needed, you are combining two resources in the same template, the idea is to create one resource (and its nested or related resources -> e.g. VirtualNetwork and Subnets) in the same template another example is VirtualMachine and NetworkInterface, NetworkInterface is always used by a VirtualMachine, so in this case we place both resources in the same template. In this specific case a Storage Account can be used by different resources (diagnostics, artifacts, store bootstrap blob containers for Palo Alto, etc.)

Reply:
removed